README.md

```diff
@@ -9,6 +9,65 @@ go get github.com/sourcegraph/sourcegraph-accounts-sdk-go
 > [!note]
 > Please submit all issues to the [`sourcegraph/sourcegraph-accounts` repository](https://github.com/sourcegraph/sourcegraph-accounts/issues) 
 
+## Authentication
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	samsauth "github.com/sourcegraph/sourcegraph-accounts-sdk-go/auth"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+)
+
+func main() {
+	stateSetter := func(w http.ResponseWriter, r *http.Request, state string) error {
+		// TODO: Save to session data. For best security measures, store in a backend
+		// component (e.g. Redis) or encrypted cookie. Unencrypted cookies are strongly
+		// discouraged as they can be tampered with.
+		return nil
+	}
+	stateGetter := func(r *http.Request) (string, error) {
+		// TODO: Retrieve from session data.
+		return "", nil
+	}
+	stateDeleter := func(w http.ResponseWriter, r *http.Request) {
+		// TODO: Delete from session data.
+	}
+
+	samsauthHandler, err := samsauth.NewHandler(
+		samsauth.Config{
+			Issuer:         "https://accounts.sourcegraph.com",
+			ClientID:       os.Getenv("SAMS_CLIENT_ID"),
+			ClientSecret:   os.Getenv("SAMS_CLIENT_SECRET"),
+			RequestScopes:  []scopes.Scope{scopes.OpenID, scopes.Email, scopes.Profile},
+			RedirectURI:    os.Getenv("SAMS_REDIRECT_URI"),
+			FailureHandler: samsauth.DefaultFailureHandler,
+			StateSetter:    stateSetter,
+			StateGetter:    stateGetter,
+			StateDeleter:   stateDeleter,
+		},
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/auth/login", samsauthHandler.LoginHandler())
+	mux.Handle("/auth/callback", samsauthHandler.CallbackHandler(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			userInfo := samsauth.UserInfoFromContext(r.Context())
```

Comment:
For clarity I'd suggest having a comment here to explain what's going on:

```go
// The SAMS auth handler will complete the login process. And if successful,
// the user will be accessible from the request context.
//
// You can assume the `userInfo` will be present. As if there were any errors
// during the login process, the user-supplied handler will not be invoked.
// (And the FailureHandler will be serve the HTTP response instead.)
```

Reply:
Another thing that I noticed when patching this locally and wiring up SAMS auth:

Seems like we are missing a `/logout` handler too... while SAMS isn't really needed on this code path, the caller is going to add whatever "local session creation" logic in the `/auth/callback`, so they would want to have some sort of "local session deletion" logic in a `/auth/logout` method as well. Right?

Longer term, if we put the SAMS login cookie on the sourcegraph.com domain, would we want `samsauth` to handle creating/loading/deleting the session? (Or perhaps that would be some other package, e.g. `samsauth_sessions.NewHandler`?

Reply:
I agree that we can add some example/docs in the future, but let's not worry about that now. 👍